packages/@azure/servicebus/data-plane/test/renewLockSessions.spec.ts

```diff
@@ -0,0 +1,715 @@
+  senderClient: QueueClient | TopicClient,
+  receiverClient: QueueClient | SubscriptionClient,
+  options: AutoLockRenewalTestOptions
+): Promise<void> {
+  let numOfMessagesReceived = 0;
+
+  await senderClient.getSender().send(testMessage);
+
+  let sessionClient = await receiverClient.getSessionReceiver({ sessionId: testSessionId });
+  await sessionClient.receive(
+    async (brokeredMessage: ServiceBusMessage) => {
+      if (numOfMessagesReceived < 1) {
+        numOfMessagesReceived++;
+
+        should.equal(brokeredMessage.body, testMessage.body);
+        should.equal(brokeredMessage.messageId, testMessage.messageId);
+
+        // Compute expected initial lock duration
+        const initialTimeUtc = new Date();
+        const expectedLockExpiryTimeUtc = new Date();
+
+        expectedLockExpiryTimeUtc.setSeconds(
+          initialTimeUtc.getSeconds() + lockDurationInMilliseconds / 1000
+        );
+
+        // Verify actual lock duration is reset
+        assertTimestampsAreApproximatelyEqual(
+          sessionClient.sessionLockedUntilUtc,
+          expectedLockExpiryTimeUtc,
+          "Initial"
+        );
+
+        // Sleeping...
+        await delay(
+          options.delayBeforeAttemptingToCompleteMessageInSeconds * 1000 +
+            lockDurationInMilliseconds +
+            1000
+        );
+
+        let errorWasThrown: boolean = false;
+        await brokeredMessage.complete().catch((err) => {
+          should.equal(err.name, "Error");
+          should.equal(!(err.message.search("Cannot find the receiver with name") + 1), false);
+          errorWasThrown = true;
+        });
+
+        should.equal(errorWasThrown, options.willCompleteFail, "Error Thrown flag value mismatch");
+      }
+    },
+    onError,
+    {
+      autoComplete: false,
+      maxAutoRenewDurationInSeconds: options.maxAutoRenewDurationInSeconds
+    }
+  );
+  await delay(options.delayBeforeAttemptingToCompleteMessageInSeconds * 1000 + 10000);
+  await sessionClient.close();
+
+  if (uncaughtErrorFromHandlers) {
+    chai.assert.fail(uncaughtErrorFromHandlers.message);
+  }
+
+  should.equal(numOfMessagesReceived, 1, "Mismatch in number of messages received");
+
+  if (options.willCompleteFail) {
+    // Clean up any left over messages
+    sessionClient = await receiverClient.getSessionReceiver({ sessionId: testSessionId });
+    const unprocessedMsgs = await sessionClient.receiveBatch(1);
+    await unprocessedMsgs[0].complete();
+  }
+}
+
+// Helper functions
+function check(str: string | undefined, label: string): string {
+  if (!str || str.trim() === "") {
+    throw new Error(`"${label}" cannot be null or empty`);
+  }
+  return str;
+}
+
+function assertTimestampsAreApproximatelyEqual(
+  actualTimeInUTC: Date | undefined,
+  expectedTimeInUTC: Date,
+  label: string
+): void {
+  if (actualTimeInUTC) {
+    should.equal(
+      Math.pow((actualTimeInUTC.valueOf() - expectedTimeInUTC.valueOf()) / 1000, 2) < 4, // Within +/- 2 seconds
```

Comment:
Change this to -

Math.pow((actualTimeInUTC.valueOf() - expectedTimeInUTC.valueOf()) / 1000, 2) < 100, // Within +/- 10 seconds

Reply:
> Math.pow((actualTimeInUTC.valueOf() - expectedTimeInUTC.valueOf()) / 1000, 2) < 100, // 

"< 10 ", right ?

Reply:
100, because the diff is being squared to take care of both +ve and -ve cases

Reply:
Oops, didn't see that!